src/ts/components/subtitletogglebutton.ts

```diff
@@ -0,0 +1,88 @@
+import { ToggleButton, ToggleButtonConfig } from './togglebutton';
+import { i18n } from '../localization/i18n';
+import { PlayerAPI, SubtitleTrack } from 'bitmovin-player';
+import { UIInstanceManager } from '../uimanager';
+import { StorageUtils } from '../storageutils';
+import { SubtitleSelectBox } from './subtitleselectbox';
+import { SettingsPanel } from './settingspanel';
+import { prefixCss } from './dummycomponent';
+import { SubtitleSwitchHandler } from '../subtitleutils';
+
+export interface StoredSubtitleLanguage {
+    language: string;
+    active: boolean;
+}
+
+export class SubtitleToggleButton extends ToggleButton<ToggleButtonConfig> {
+    private settingsPanel: SettingsPanel;
+    private subtitleSelectBox: SubtitleSelectBox;
+    private player: PlayerAPI;
+
+    constructor(subtitleSettingsOpenButton: SettingsPanel, subtitleSelectBox: SubtitleSelectBox, config: ToggleButtonConfig = {}) {
+        super(config);
+
+        this.settingsPanel = subtitleSettingsOpenButton;
+        this.subtitleSelectBox = subtitleSelectBox;
+
+        const defaultConfig: ToggleButtonConfig = {
+            cssClass: 'ui-subtitletogglebutton',
+            onClass: 'subtitles-on',
+            offClass: 'subtitles-off',
+            text: i18n.getLocalizer('settings.subtitles'),
+            ariaLabel: i18n.getLocalizer('settings.subtitles'),
+        };
+
+        this.config = this.mergeConfig(config, defaultConfig, this.config);
+    }
+
+    configure(player: PlayerAPI, uimanager: UIInstanceManager): void {
+        super.configure(player, uimanager);
+        this.player = player;
+        this.hide();
+        this.initPlayerEvents();
+
+        this.onClick.subscribe(() => {
+            const availableSubtitles = player.subtitles.list();
+            const storedSubtitle: StoredSubtitleLanguage = StorageUtils.getObject(prefixCss('subtitlelanguage'));
+            const subtitleTrack = storedSubtitle ? availableSubtitles.find(e => e.lang === storedSubtitle.language) : undefined;
+            if (!subtitleTrack) {
+                this.settingsPanel.show();
+            } else if (this.isOff() && subtitleTrack) {
+                this.on();
+                player.subtitles.enable(subtitleTrack.id);
+                SubtitleSwitchHandler.setSubtitleLanguageStorage(this.player, subtitleTrack.id);
+            } else {
+                this.off();
+                player.subtitles.disable(subtitleTrack.id);
+                SubtitleSwitchHandler.setSubtitleLanguageStorage(this.player);
```

Comment:
Since the bindSelectionEvent already handles all subtitle changes... do we even need the calls to `SubtitleSwitchHandler.setSubtitleLanguageStorage` here or would it also work without them?

Reply:
Tried it out... does not work without